vision_evaluation/evaluators.py

```diff
@@ -1100,3 +1100,46 @@ def add_predictions(self, predictions, ground_truths):
 
     def get_report(self, **kwargs):
         return {self._get_id(): {group: eval.get_report(**kwargs) for group, eval in self._evaluators.items()}}
+
+
+class MeanLpErrorEvaluator(Evaluator):
+    """
+    Error evaluator for regression targets
+    Parameters:
+        p: Lp order of the norm used
+    """
+
+    def __init__(self, p: int):
+        assert p != 0
+        self.p = p
+
+        self.total_num = 0
+        self.total_error = 0
+
+        super(MeanLpErrorEvaluator, self).__init__()
+        
+    def reset(self):
+        super(MeanLpErrorEvaluator, self).reset()
+        self.total_num = 0
+        self.total_error = 0
+
+    def _get_id(self):
+        return f'l{self.p}'
+
+    def add_predictions(self, predictions, targets):
+        """ Evaluate a batch of predictions.
+        Args:
+            predictions: the model output numpy array. Shape (N,)
+            targets: the golden truths. Shape (N,)
+        """
+        assert len(predictions) == len(targets)
+        assert len(targets.shape) == 1
+
+        n_sample = len(predictions)
+
+        self.total_error += np.linalg.norm(predictions - targets, ord=self.p)**self.p
```

Comment:
why not

sum(np.power(predictions - targets), self.p))

Reply:
Good point, added np.abs also. Thanks!